app/character/knight.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+from app.character.inventory import Inventory
+
+
+class Knight:
+    def __init__(self,
+                 k_name: str,
+                 k_power: int = 1,
+                 k_hp: int = 1,
+                 k_armors=[],
+                 k_weapon=None,
+                 k_potion=None
+                 ) -> None:
+        self.name = k_name
+        self.power = k_power
+        self.hp = k_hp
+        self.protection = 0
+        self.items = Inventory(k_weapon, k_armors, k_potion)
+        self.is_ready = False
+        self.is_armored = False
+        self.is_weapon_up = False
+        self.effect = None
+
+    @classmethod
+    def from_dict(cls, pers: dict) -> Knight:
+        return cls(
+            pers["name"],
+            pers["power"],
+            pers["hp"],
+            pers["armour"],
+            pers["weapon"],
+            pers["potion"]
+        )
+
+    def use_potion(self):
+        if self.items.potion:
+            print(f"{self.name} drinking potion {self.items.potion.name}")
+            boost = self.items.potion.effect
+            boost_k = boost.keys()
+            if "hp" in boost_k:

```

Comment:
These code duplications can be fixed, using `stats = ["protection", "power", "hp"]` and looping on `stats`

Reply:
> These code duplications can be fixed, using `stats = ["protection", "power", "hp"]` and looping on `stats`

How to loop `self.hp`, `self.power` and `self.protection`?
I can create a dict, but i think it can be less readability